### ISSUE TITLE
[v2 QUESTION]: Chemprop Training with Full Dataset

### ISSUE THREAD

**Issue by mmahmedgithub** [chemprop/chemprop#1266]
Hi,

I’m looking for advice on the best approach to train a Chemprop v2 model using the entire dataset for a production-ready deployment.

I've already used my dataset for hyperparameter optimization and training, leveraging my own split column. Therefore, I have a clear idea of expected performance, optimal hyperparameters, and anticipated real-world performance based on those previous phases.

However, when attempting to train using 100% of the dataset (no splits), Chemprop v2 raises errors about missing test/validation sets or insufficient data, even when specifying splits like [0.99, 0.01, 0.00]. Given my limited data, it's critical to include every datapoint in training the final production model.

Is there a recommended workaround or an easy solution for this issue?

Thanks,
Marawan

**Comment by KnathanM** [comment]
Hi @mmahmedgithub great question. We are working on making the CLI work with no splits, but that hasn't been merged in yet. In the mean time what I do as a work around is copy the last row in the data csv four times to the end and specify those as the val and test set. (We need two data points in a dataset in case batch normalization is being used. This is something we could improve in the future.) I think in this case the easiest way to specify the throwaway val and test sets is to use a splits column. 

So if your csv looks like:
| smiles | y |
|--------|---|
| C      | 1 |
| CC     | 2 | 
| CCC    | 3 | 

you can modify it to look something like this:
| smiles | y | split |
|--------|---|-------|
| C      | 1 | train |
| CC     | 2 | train |
| CCC    | 3 | train |
| CCC    | 3 | val   |
| CCC    | 3 | val   |
| CCC    | 3 | test  |
| CCC    | 3 | test  |

and your command something like `chemprop train -i data.csv --smiles-columns smiles --target-columns y --splits-column split`

The val and test loop will run with those duplicate points, but you can ignore the predictions file that is generated. The model you want will be called `last.ckpt` in the checkpoint file. Importantly this is not `best.pt` which is determined based on the validation set.
